src/api/environment.cc

```diff
@@ -266,6 +307,15 @@ void SetIsolateMiscHandlers(v8::Isolate* isolate, const IsolateSettings& s) {
 
   if (s.flags & DETAILED_SOURCE_POSITIONS_FOR_PROFILING)
     v8::CpuProfiler::UseDetailedSourcePositionsForProfiling(isolate);
+
+  if ((s.flags & HAS_ATOMICS_WAIT_CALLBACK) &&
+      s.atomics_wait_callback != nullptr) {
+    isolate->SetAtomicsWaitCallback(
+        s.atomics_wait_callback, s.atomics_wait_callback_data);
+  } else {
+    isolate->SetAtomicsWaitCallback(
```

Comment:
Do we know the effect on perf from always setting this?

Reply:
I can try to measure it but I’m assuming that it’s small compared to the actual work that V8 does for `Atomics.wait()`.

Reply:
Fwiw, the performance impact *is* measurable, with basically all of it coming from the extra `CreationContext()` call. I’m trying to figure out ways around that, but in any case, thanks for drawing my attention to this.

Reply:
@devsnek I’ve switched this so that the hook only is installed when the option is set, similar to what we do for `--trace-uncaught`. There is no noticeable performance difference in the default case anymore now.